### DISCUSSION TITLE
AT command slave

### DISCUSSION THREAD

**Discussion by dzurikmiroslav** [dzurikmiroslav/esp32-evse#22] [Polls]
Should the firmware support AT command slave on RS485 when it supports the MODBUS protocol? (for example: controlling EVSE from Raspberry PI) Currently the AT api is not very well maintained, I'm thinking about removing it...

**Comment by jetpax** [comment]
You mean removing libcat support?

This could be useful for 4G modem modules that use AT commands

**Reply by dzurikmiroslav** [reply]
I guess I asked the wrong question, I meant AT command slave mode. The cAT library is very good, but it is only suitable for implementing a slave device. 
Yes, direct 4G modem support would be great, but that's a different feature. This feature is called PPP, ESP-IDF also supports it https://docs.espressif.com/projects/esp-protocols/esp_modem/docs/latest/README.html

**Reply by jetpax** [reply]
My bad, thanks for explaining that.  I would say let it go. It would be possible for the Pi to use the REST API no?

Thanks also for pointing out the ESP modem support, good to see it includes CMUX.

**Reply by dzurikmiroslav** [reply]
I recommend using Modbus RTU on the serial line, there is available registers https://github.com/dzurikmiroslav/esp32-evse/wiki/Modbus#table-of-modbus-registers
Through network is able MQTT, Modbus TCP or REST api
